templates/arbiter_outputs_entry.py

```diff
@@ -0,0 +1,247 @@
+        )
+        self.in_use_counters.append(in_use_counter)
+        self.in_uses.append(0)
+        self.is_connected.append(0)
+        in_use_dest.connect()
+        in_use_conn_dest.connect()
+        self._channels.append(in_use_dest)
+        self._channels.append(in_use_conn_dest)
+        ic_ch = in_use_counter.create_channel()
+        ic_ch.connect()
+        self._channels.append(ic_ch)
+
+        # Combine the ok and in_use channels to get the fault counts
+        # We are fauling if ok=False and in_use=True
+        # Summarize this as a combined local channel
+        fault_summary = FaultSummary(
+            ok_pvname=f"ca://{self.prefix}FFO:{self.ffo}:FF:{fault_num_str}:OK_RBV",
+            in_use_pvname=f"ca://{self.prefix}FFO:{self.ffo}:FF:{fault_num_str}:Info:InUse_RBV",
+            parent=self,
+        )
+        fault_summary_ch1, fault_summary_ch2 = fault_summary.create_channels()
+        fault_summary_ch3 = PyDMChannel(
+            address=f"loc://{self.prefix}{self.ffo}{fault_num_str}:SUMMARY?type=int&init=0",
+            value_signal=fault_summary.value_sig,
+            value_slot=self.show_loc_connected,
+        )
+        fault_counter = CounterElement(
+            pvname=f"loc://{self.prefix}{self.ffo}{fault_num_str}:SUMMARY?type=int&init=0",
+            index=self.loop_count,
+            value_cache=self.faults,
+            parent=self,
+        )
+        fault_counter_ch = fault_counter.create_channel()
+        fault_dest_ch = PyDMChannel(
+            address=f"loc://{self.prefix}{self.ffo}:FaultCount?type=int&init=0",
+            value_slot=self.new_fault_count,
+            value_signal=fault_counter.value_sig,
+        )
+        self.fault_summaries.append(fault_summary)
+        self.fault_counters.append(fault_counter)
+        self.faults.append(0)
+        # Connect from the last in the chain to the first
+        for ch in (
+            fault_dest_ch,
+            fault_counter_ch,
+            fault_summary_ch3,
+            fault_summary_ch2,
+            fault_summary_ch1,
+        ):
+            ch.connect()
+            self._channels.append(ch)
+
+        self.loop_count += 1
+        self.fault_num += 1
+
+        if self.fault_num <= self.ff_end:
+            self.next_counter_soon()
+
+    def show_loc_connected(self, *args, **kwargs):
+        ...
+
+    def new_fault_count(self, count: int):
+        if count:
+            self.ui.fault_label.alarm_severity_changed(2)
+        else:
+            self.ui.fault_label.alarm_severity_changed(0)
+
+    def new_bypass_count(self, count: int):
+        if count:
+            self.ui.bypass_label.alarm_severity_changed(1)
+        else:
+            self.ui.bypass_label.alarm_severity_changed(0)
+
+    def channels(self):
+        return self._channels
+
+    def ui_filename(self):
+        return 'arbiter_outputs_entry.ui'
+
+
+class CounterElement(QObject):
+    value_sig = Signal(int)
```

Comment:
I like this pattern - it's kinda like a self-contained ophyd device but with components as Qt signals